### DISCUSSION TITLE
Inspiration for a coding-task that I was sent

### DISCUSSION THREAD

**Discussion by beschmitt** [neuefische/questions#768] [Data Analytics]
Hey all,

I'm currently in the application process for a Data Analyst-position. For the next interview I have to prepare a presentation based on data and a task that the company sent me.

They want me to analyse data and based on it find trends and make recommendations for the company. However I'm having problems since the entire dataframe for that task consists of three columns:
-How many times a customer has ordered from our store
-How much money that customer has spent overall
-The average number of days between each order

Based on this I've calculated how much each customer has spent for an order on average and I also grouped the data by the number of orders to see if more orders automatically means more money spent.
However I'm struggling to find other things to calculate and I don't know what recommendations I could make based on the data that I have.

Thank you in advance.

**Comment by CStolz18** [comment]
Hi,

just a couple of ideas that may work:

- you could segment customers based on their ordering behaviour. E.g.:
   New customers (low order count), regular customers (medium order count), loyal customers (high order count)
- group customers based on the month or year of their (first) order. This can help you track how customer behaviour changes over time
- calculate the percentage of customers who have made repeat purchases. This could help to find out about customer satisfaction/loyalty
- you could also calculate the % of customers who haven't made a purchase in a certain time frame to identify at-risk customers

Possible recommendations:

- tailor marketing strategies based on customer segmentation (for new customers, focus on attracting them back with promotions. For loyal customers, consider loyalty programs or exclusive offers, etc.)

- implement a churn prevention strategy for customers who haven't made a purchase in a while. They could send re-engagement emails with incentives, conduct customer surveys, or offer support to address any issues they might have.

Hope, it will help to find some ideas

**Reply by beschmitt** [reply]
The first point with the segmenting is a good idea. Is was also thinking about clustering them based on amount of orders and money spent to see if there are possible groups.

Unfortunately I can't do anything with time-data because I don't have that. I literally only have how many times that customer has ordered in the last five years. And all customers have ordered at least two times, so there are no customers without repeat purchases.

**Comment by charanm89** [comment]
Hello,
Some other solutions could be

- Using Recency, Frequency, Monetary (RFM) analysis. While you have information on order frequency and monetary value, you can segment customers based on recency, which is a measure of how recently they made a purchase. This can provide insights into which customers are more active.

- If all customers are having minimum 2 purchases as you say, you could calculate churn rate, maybe if they are not making a 3rd purchase, you are losing that customer, and could have targetted marketing to get them back
- You could make some assumptions on lifetime of the customer based on the industry of the company, lets say 3yrs, calculate the avg order value, then multiply with the lifetime assumption to get Assumed CLV (customer life time value). Here there might be industry averages so you could create some recommendations to increase the value if low.
- If there is trend of periodicity to the gap between each order, you could also suggest a subscription tier that could be beneficial.

Hope this helps
